rats-tls/src/attesters/sev/ttrpc/src/lib.rs

```diff
@@ -0,0 +1,135 @@
+extern crate sev;
+
+use anyhow::{Context, Result};
+use protocols::sev::{aeb, aeb_ttrpc};
+use serde_json;
+use sev_evidence::SevEvidence;
+use ttrpc::client::Client;
+
+mod protocols;
+mod sev_evidence;
+
+#[macro_use]
+extern crate log;
+
+const SERVER_ADDR: &str = "vsock:///tmp/aeb.sock";
+const SERVER_PORT: &str = "5577";
+
+#[no_mangle]
+pub extern "C" fn retrieve_attestation_evidence(
+    guest_handle: u32,
+    evidence_size: u32,
+) -> Option<Box<SevEvidence>> {
+    let req = aeb::RetrieveAttestationEvidenceRequest {
+        guest_handle: guest_handle,
+        evidence_size: evidence_size,
+        unknown_fields: Default::default(),
+        cached_size: Default::default(),
+    };
+
+    let agent_client = connect().context("connect");
+    match agent_client {
+        Ok(agent_client) => match agent_client.retrieve_attestation_evidence(default_ctx(), &req) {
+            Ok(res) => {
+                let evidence = res.get_evidence();
+                let wrap_command = String::from_utf8(evidence.to_vec())
+                    .and_then(|evidence| Ok(serde_json::from_str::<SevEvidence>(&evidence[..])))
+                    .unwrap();
+
+                match wrap_command {
+                    Ok(wrap_command) => {
+                        debug!("sev evidence is {:#X?}", wrap_command);
+                        return Some(Box::new(wrap_command));
+                    }
+                    Err(e) => {
+                        error!("failed to deserialize aeb respense to sev evidence {:?}", e);
+                        return None;
+                    }
+                }
+            }
+            Err(e) => {
+                error!("failed to retrieve evidence from aeb service {:?}", e);
+                return None;
+            }
+        },
+        Err(e) => {
+            error!("failed to create ttrpc client {:?}", e);
+            None
+        }
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn retrieve_attestation_evidence_size(guest_handle: u32) -> u32 {
+    let req = aeb::RetrieveAttestationEvidenceSizeRequest {
+        guest_handle: guest_handle,
+        unknown_fields: Default::default(),
+        cached_size: Default::default(),
+    };
+
+    let agent_client = connect().context("connect");
```

Comment:
Maybe you can change the following snippet codes like this:
let res = connect().context("connect")
    .map_err(|e| {
	    error!("failed to retrieve evidence size from aeb service {:?}", e);
	    0
	})
	.and_then(|agent_client| {
	    let res = agent_client.retrieve_attestation_evidence_size(default_ctx(), &req)
		    .map_err(|e| {
			    error!("failed to retrieve evidence size from aeb service {:?}", e);
			    0
			})
			.and_then(|res| {res.evidence_size})
		OK(res)
	})
	.unwrap();